src/matching/team_recommendations.py

```diff
@@ -13,127 +36,84 @@ def get_team_recommendations(email):  # GET
-        for match in complementary_skills_match:
-            if match['_id'] not in names:
-                names.add(match['_id'])
-                matches.append(match)
-
-    # add base on interests
-    # AR/VR, BlockChain, Communications, CyberSecurity, DevOps, Fintech, Gaming,
-    # Healthcare, IoT, LifeHacks, ML/AI, Music, Productivity, Social Good, Voice Skills
-
-    # finding team with listed interests, if too much matches, find from teams in the matches
-    if len(matches) > 50:
-        for match in matches:
-            if len(matches) <= 50:
-                break
-            team_interests = match["meta"]["interests"]
-            # team has no common skill
-            if len(list(set(interests).intersection(set(team_interests)))) == 0:
-                matches.remove(match)
-                names.remove(match["_id"])
+        else:
+            needed_skills.update(skill_set)
     else:
-        for interest in interests:
-            match = coll("teams").aggregate([{"$match": {"complete": False, "meta.interest": {"$all": [interest]}}}])
-            if not match:
-                continue
-            for m in match:
-                if m["_id"] not in names:
-                    names.add(m["_id"])
-                    matches.append(m)
-
-    # add suggestions base on prize
-    for prize in prizes:
-        match = coll("teams").aggregate([{"$match": {"complete": False, "prizes": {"$all": [prize]}}}])
-        if not match:
-            continue
-        for m in match:
-            if m["_id"] not in names:
-                names.add(m["_id"])
-                matches.append(m)
-
-    # if there are too many matches, reduce it base on seriousness
-    if len(matches) > 20:
-        for team in matches:
-            if (abs(team["seriousness"] - seriousness)) > 2:
-                matches.remove(team)
-                names.remove(team["_id"])
-
-    # current_team = coll("teams").find_one({"_id": user["team_id"]})
-    # try:
-    #     matches.remove(current_team)
-    # except ValueError:
-    #     pass
-
-    # inv_in = current_team["incoming_inv"]
-    # inv_out = current_team["outgoing_inv"]
-
-    # inv_sum = set()
-    # inv_sum.update(set(inv_in))
-    # inv_sum.update(set(inv_out))
-
-    # for i in inv_sum:
-    #     try:
-    #         matches.remove(i)
-    #     except ValueError:
-    #         pass
+        if front_pers > 0.3:
+            needed_skills.update(backend_languages)
+        else:
+            needed_skills.update(skill_set)
+
+    for team_id in team_map:
+        target_team = coll("teams").find_one({"_id": team_id})
+        target_team_skills = target_team["skills"]
+        intersection_size = len(set(target_team_skills).intersection(needed_skills))
+        team_map[team_id] *= (intersection_size * skills_weight)
+        team_seriousness = target_team["meta"]["seriousness"]
+        team_map[team_id] = team_map[team_id] * (intersection_size * skills_weight) * \
```

Comment:
Having weights makes sense, but we might want to run a few test scenarios and see how much it actually affects things. For example, we would not want to recommend seriousness 1 to seriousness 5 - that should be a pretty significant difference, meaning more weight.